### PR TITLE
Refactoring variables + tflint

### PR DESCRIPTION
**Pull request by filippolmt**

### CODE REVIEW

File: values/kube-prometheus-stack.yml [hunk 1]
```diff
@@ -19,18 +19,18 @@ grafana:
     enabled: ${grafana_ingress_enabled}
     annotations:
       kubernetes.io/ingress.class: ${ingress_class}
-      %{~ if cert_manager_enabled == 1 ~}
-      cert-manager.io/cluster-issuer: "${cluster_issuer_name}"
+      %{~ if grafana_cert_manager_enabled == 1 ~}
```
Comment (Monska85) [inline values/kube-prometheus-stack.yml:22]: I think that this variable is a boolean so we can use this directly

File: values/kube-prometheus-stack.yml [hunk 2]
```diff
@@ -19,18 +19,18 @@ grafana:
...
+      cert-manager.io/cluster-issuer: "${grafana_cluster_issuer_name}"
       acme.cert-manager.io/http01-edit-in-place: "true"
       %{~ endif ~}
     hosts:
-      - ${ingress_host}
-      %{~ if cert_manager_enabled == 1 ~}
+      - ${grafana_ingress_host}
+    %{~ if cert_manager_enabled == 1 ~}
```
Comment (Monska85) [inline values/kube-prometheus-stack.yml:28]:
```suggestion
    %{~ if grafana_cert_manager_enabled ~}
```

File: variables.tf [hunk 3]
```diff
@@ -1,61 +1,35 @@
...
-variable "grafana_ingress_enabled" {
-  type = bool
-  default = true
-}
-
-variable "adapter_chart_version" {
+variable "prometheus_adapter_chart_version" {
   type = string
+  description = "Chart version Prometheus Adapter"
   default = ""
```
Comment (Monska85) [inline variables.tf:27]: I think that we should not define a default. Without a default the error will be raised for `not defined variable` and not when the helm provider will try to install the chart

File: variables.tf [hunk 4]
```diff
@@ -1,61 +1,35 @@
...
 variable "regcred" {
   type = string
-}
-
-variable "secret_name" {
-  type = string
+  description = "Name of the secret docker credential"
 }
 
 variable "storage_class_name" {
```
Comment (Monska85) [inline variables.tf:35]: Add a description. Maybe something like:

```yaml
description = "Storage class name which will be used for the `PersistentVolumes` for Prometheus and Grafana"
```

File: variables.tf [hunk 5]
```diff
@@ -66,59 +40,101 @@ variable "storage_class_name" {
...
+variable "grafana_ingress_basic_auth_message" {
+  type = string
+  description = "Grafana basic auth message."
+  default = "Authentication Required"
+}
+
+variable "grafana_cert_manager_cluster_issuer_name" {
+  type = string
+  description = "Resource representing the cluster issuer of cert-manager (used to deploy a TLS cert for Grafana ingress)."
+  default = ""
```
Comment (Monska85) [inline variables.tf:118]: I think that we should not define a default. Without a default the error will be raised for `not defined variable` and not when the helm provider will try to install the chart.

Maybe we can be more explicit about the behaviour of this variable: if this variable is left empty the `cert-manager` annotations will not be put in place in the `Ingress`.

File: variables.tf [hunk 6]
```diff
@@ -66,59 +40,101 @@ variable "storage_class_name" {
 variable "prometheus_resources" {
   type        = any
```
Comment (Monska85) [inline variables.tf:41]: I would prefer to define the object structure using the following syntax:

```suggestion
  type        = object({ cpu_requests = string, memory_requests = string })
```

File: variables.tf (hunk 3 above)
Comment (Monska85) [reply variables.tf:21]: Maybe we can be more explicit about the behaviour of this variable: if this variable is left empty the Prometheus Adapter will not be installed.

File: .tflint.hcl [hunk 7]
```diff
@@ -0,0 +1,7 @@
+config {
+  varfile = ["examples/test.tfvars"]
+}
+
+rule "terraform_unused_declarations" {
+  enabled = true
+}
```
Comment (paolomainardi) [inline .tflint.hcl:7]: We should add more rules from here: https://github.com/terraform-linters/tflint/blob/master/docs/rules/README.md

I suggest:

* terraform_comment_syntax
* terraform_deprecated_index
* terraform_deprecated_interpolation	
* terraform_module_pinned_source	
* terraform_module_version	
* terraform_naming_convention
* terraform_standard_module_structure		
* terraform_typed_variables	
* terraform_unused_required_providers	
* 



<br class="Apple-interchange-newline">[terraform_comment_syntax](https://github.com/terraform-linters/tflint/blob/master/docs/rules/terraform_comment_syntax.md)

File: README.md [hunk 8]
```diff
@@ -23,20 +29,23 @@ provider "helm" {
 ```
 module "kube_prometheus_stack" {
   source = "sparkfabrik/terraform-sparkfabrik-prometheus-stack"
+  prometheus_stack_chart_version = "v31.0.0"
```
Comment (Monska85) [inline README.md:32]:
```suggestion
  prometheus_stack_chart_version = "31.0.0"
```

File: examples/main.tf [hunk 9]
```diff
@@ -1,18 +1,23 @@
...
-  chart_version = "v30.2.0"
-  ingress_host = "monitoring.example.com"
-  basic_auth_username = "admin"
-  cluster_issuer_name = "prod-certmanager"
-  install_adapter = "true"
-  adapter_chart_version = "3.0.1"
-  regcred = "true"
-  pull_secrets = "docker"
-  secret_name = "monitoring-tls"
+  prometheus_stack_chart_version = "v31.0.0"
```
Comment (Monska85) [inline examples/main.tf:3]:
```suggestion
  prometheus_stack_chart_version = "31.0.0"
```

File: examples/test.tfvars [hunk 10]
```diff
@@ -0,0 +1,34 @@
+chart_version = "v30.2.0"
```
Comment (Monska85) [inline examples/test.tfvars:1]:
```suggestion
prometheus_stack_chart_version = "31.0.0"
```

**Comment by Monska85** [comment]
Try also to apply the linter suggestions:

- change `value = "${var.namespace}/${kubernetes_secret.kube_prometheus_ingress_auth.0.metadata.0.name}"` with `value = "${var.namespace}/${kubernetes_secret.kube_prometheus_ingress_auth.[0].metadata.[0].name}"`
- rename the file `output.tf` in `outputs.tf`

I would prefer to keep all the variables in all places (definition, examples, module declaration, readme, ...) in the same order, what do you think about this?

File: README.md (hunk 8 above)
Comment (filippolmt) [reply README.md]: Done

File: examples/main.tf (hunk 9 above)
Comment (filippolmt) [reply examples/main.tf]: Done

File: .tflint.hcl (hunk 7 above)
Comment (filippolmt) [reply .tflint.hcl]: Added, with the exception of terraform module version, rule not found: https://github.com/sparkfabrik/terraform-sparkfabrik-prometheus-stack/runs/5051758323?check_suite_focus=true

File: examples/test.tfvars (hunk 10 above)
Comment (filippolmt) [reply examples/test.tfvars]: Done

File: README.md [hunk 11]
```diff
@@ -23,20 +29,23 @@ provider "helm" {
 ```
 module "kube_prometheus_stack" {
   source = "sparkfabrik/terraform-sparkfabrik-prometheus-stack"
+  prometheus_stack_chart_version = "31.0.0"
```
Comment (Monska85) [inline README.md:32]: Sorry, I didn't catch the version mismatch in the latest review

```suggestion
  prometheus_stack_chart_version = "32.0.0"
```

**Pull request merged**